src/NServiceBus.Core/Performance/Statistics/SLA/SLAMonitoring.cs

```diff
@@ -104,26 +102,40 @@ protected override Task OnStop(IMessageSession session)
 
             void UpdateTimeToSLABreach()
             {
-                List<DataPoint> snapshots;
+                var snapshots = new DataPoint[MaxDataPoints];
 
                 lock (dataPoints)
                 {
-                    snapshots = new List<DataPoint>(dataPoints);
+                    Array.Copy(dataPoints, snapshots, MaxDataPoints);
                 }
 
                 var secondsToSLABreach = CalculateTimeToSLABreach(snapshots);
 
                 counter.RawValue = Convert.ToInt32(Math.Min(secondsToSLABreach, int.MaxValue));
             }
 
-            double CalculateTimeToSLABreach(List<DataPoint> snapshots)
+            double CalculateTimeToSLABreach(DataPoint[] snapshots)
             {
+                if (snapshots.Length == 0)
```

Comment:
It will never be zero now. How we should work with array with `default` elements though?